src/pages/home/components/nav-bar.tsx

```diff
@@ -1,10 +1,36 @@
 import { User } from 'networking/types/user';
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+import { useDebounce } from 'hooks/useDebounce';
 import { AppLink, RouteName } from 'routes';
+import { UserContext } from '../../../common/user-context';
 import styles from './nav-bar.module.scss';
 
 const NavBar = () => {
   const activeUser = JSON.parse(localStorage.getItem('activeUser') ?? '') as User;
+  const contextValue = useContext(UserContext);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const { defaultUsers } = contextValue;
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 500);
+
+  useEffect(() => {
+    if (debouncedSearchTerm) {
+      // eslint-disable-next-line max-len
+      const filter = contextValue.defaultUsers.filter((user) => user.firstName.toUpperCase().match(debouncedSearchTerm.toUpperCase()));
+
+      if (!filter.length) {
+        contextValue.updateFilterUsers(defaultUsers);
+      } else {
+        contextValue.updateFilterUsers(filter);
+      }
+    } else {
+      contextValue.updateFilterUsers(defaultUsers);
+    }
+  }, [debouncedSearchTerm]);
```

Comment:
have you heard about DRY?
```suggestion
    if (debouncedSearchTerm) {
      // eslint-disable-next-line max-len
      const filter = contextValue.defaultUsers.filter((user) => user.firstName.toUpperCase().match(debouncedSearchTerm.toUpperCase()));

    to_filter = (filter.length) ? filter : defaultUsers;
    contextValue.updateFilterUsers(to_filter);
  }, [debouncedSearchTerm]);
```